tests/pull.bats

```diff
@@ -167,3 +167,60 @@ load helpers
+  run_buildah pull --signature-policy ${TESTSDIR}/policy.json --decryption-key ${TESTDIR}/tmp/mykey.pem oci:${TESTDIR}/tmp/busybox_enc
+
+  rm -rf ${TESTDIR}/tmp
+}
+
+@test "pull encrypted registry image" {
+  _prefetch busybox
+  mkdir ${TESTDIR}/tmp
+  openssl genrsa -out ${TESTDIR}/tmp/mykey.pem 1024
+  openssl genrsa -out ${TESTDIR}/tmp/mykey2.pem 1024
+  openssl rsa -in ${TESTDIR}/tmp/mykey.pem -pubout > ${TESTDIR}/tmp/mykey.pub
+  run_buildah push --signature-policy ${TESTSDIR}/policy.json --tls-verify=false --creds testuser:testpassword --encryption-key jwe:${TESTDIR}/tmp/mykey.pub busybox docker://localhost:5000/buildah/busybox_encrypted:latest
+
+  # Try to pull encrypted image without key should fail
+  run_buildah 125 pull --signature-policy ${TESTSDIR}/policy.json --tls-verify=false --creds testuser:testpassword docker://localhost:5000/buildah/busybox_encrypted:latest
+  # Try to pull encrypted image with wrong key should fail
+  run_buildah 125 pull --signature-policy ${TESTSDIR}/policy.json --tls-verify=false --creds testuser:testpassword --decryption-key ${TESTDIR}/tmp/mykey2.pem docker://localhost:5000/buildah/busybox_encrypted:latest
+  # Providing the right key should succeed
+  run_buildah pull --signature-policy ${TESTSDIR}/policy.json --tls-verify=false --creds testuser:testpassword --decryption-key ${TESTDIR}/tmp/mykey.pem docker://localhost:5000/buildah/busybox_encrypted:latest
+  run_buildah rmi localhost:5000/buildah/busybox_encrypted:latest
+
+  rm -rf ${TESTDIR}/tmp
+}
+
+@test "pull encrypted registry image from commit" {
+ _prefetch busybox
+  mkdir ${TESTDIR}/tmp
+  openssl genrsa -out ${TESTDIR}/tmp/mykey.pem 1024
+  openssl rsa -in ${TESTDIR}/tmp/mykey.pem -pubout > ${TESTDIR}/tmp/mykey.pub
+  run_buildah from --quiet --pull=false --signature-policy ${TESTSDIR}/policy.json busybox
+  cid=$output
+  run_buildah commit --iidfile /dev/null --tls-verify=false --creds testuser:testpassword --signature-policy ${TESTSDIR}/policy.json --encryption-key jwe:${TESTDIR}/tmp/mykey.pub -q $cid docker://localhost:5000/buildah/busybox_encrypted:latest
+
+  # Try to pull encrypted image without key should fail
+  run_buildah 125 pull --signature-policy ${TESTSDIR}/policy.json --tls-verify=false --creds testuser:testpassword docker://localhost:5000/buildah/busybox_encrypted:latest
+  # Try to pull encrypted image with wrong key should fail
+  run_buildah 125 pull --signature-policy ${TESTSDIR}/policy.json --tls-verify=false --creds testuser:testpassword --decryption-key ${TESTDIR}/tmp/mykey2.pem docker://localhost:5000/buildah/busybox_encrypted:latest
+  # Providing the right key should succeed
+  run_buildah pull --signature-policy ${TESTSDIR}/policy.json --tls-verify=false --creds testuser:testpassword --decryption-key ${TESTDIR}/tmp/mykey.pem docker://localhost:5000/buildah/busybox_encrypted:latest
+  run_buildah rmi localhost:5000/buildah/busybox_encrypted:latest
+
+  rm -rf ${TESTDIR}/tmp
+}
```

Comment:
I tried building a public/private key with a passphrase.  The push went ok, but when I pulled, I got the below error.  Is there a way to include the passphrase for the private key?

```
# buildah push --encryption-key jwe:public.pem alpine docker://localhost:5000/alpine_encrypted1:latest

# buildah pull --decryption-key private.pem docker://localhost:5000/alpine_encrypted:latest
unable to obtain decrypt config: invalid decryption keys: : Missing password for encrypted private key
```

Reply:
Apparently this is implemented using the Go openpgp package.

And there seems to be a syntax for passing passphrases, looking at https://github.com/containers/ocicrypt/blob/bdddf7ce0bed9bead019e491cc7e781d5c7d8c4b/helpers/parse_helpers.go#L100-L107 . That definitely should have user-facing documentation _somewhere_, and the man pages should point to it.

Reply:
Yup - I will update the documentation. Just tested locally:

```
# Without passphrase
vagrant@ubuntu-bionic:~/go/src/github.com/containers/buildah$ sudo ./buildah --tls-verify=false pull --decryption-key privkey.pem docker://localhost:5000/alpine_encrypted1:latest
unable to obtain decrypt config: invalid decryption keys: : Missing password for encrypted private key

# With passphrase
vagrant@ubuntu-bionic:~/go/src/github.com/containers/buildah$ sudo ./buildah --tls-verify=false pull --decryption-key privkey.pem:foobar docker://localhost:5000/alpine_encrypted1:latest
Getting image source signatures
Copying blob 3c83cf2f9e96 done
Copying config af88fdb253 done
Writing manifest to image destination
Storing signatures
af88fdb253aac46693de7883c9c55244327908c77248d7654841503f744aae8b```

Reply:
Updated the docs to: 


**--decryption-key** *key*

The [keyfile[:passphrase]] to be used for decryption of images. Keyfiles can point to keys and/or certificates.       Decryption will be tried with all keys. If the key is protected by a passphrase is required to be paseed in, and      should be ommitted otherwise.